Fixed minor bug parsing url connection. Updated dependencies libraries.

ja-jimenez opened the pull request:
Hi!

I have found the url parser didn't work with url like **"mongodb://localhost/"**, so I have made some changes in the regular expression to support this and other cases like options format following the  mongodb [specs](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst). I have added tests from this spec too.

Besides, I have update some libraries to avoid some warnings while compiling and add a **@tag :rs_required** to some tests that need a replica set to be executed correctly, like the ones that use transactions.

Regards.

zookzook commented:
Thank you. It seems by updating the credo package a new rule was introduced and the credo task returns a ' Refactoring opportunities'. Please change the value to 4 or disable the rule to continue the workflow.

ja-jimenez commented:
Hi.

I have updated the credo config file to allow max_nesting value to 4. Now it must return the same output than previous version. 

Looks like the function **ReadPreference.primary()** has been deleted in master branch. I think now we must use the function **ReadPreference.merge_default()**, so I have updated the test **topology_test.exs** file to use this new function.

Regards.

ja-jimenez commented:
Hello.

Why is failing the **release-please** action?? Reviewing the logs I see this message: 

`Error: release-please failed: Error adding to tree: 9e4d56957a8bd2605bf50dbe5eea551b4fc2dc40`

Is there something that I must do to fix this?

Regards.
